File: interop/packages/dart/test/smoke_test.dart
import 'package:kosher_rust_for_dart/kosher_rust_for_dart.dart';
import 'package:test/test.dart';

void main() {
  group('kosher_rust_for_dart smoke', () {
    test('gregorian/hebrew round trip via Calendar', () {
      final calendar = Calendar();
      final civil = CivilDate(year: 2024, month: 1, day: 20);

      final hebrew = calendar.gregorianToHebrew(civil);
      expect(hebrew, isNotNull);
      expect(hebrew!.year, 5784);
      expect(hebrew.month, 5);
      expect(hebrew.day, 10);

      final back = calendar.hebrewToGregorian(hebrew);
      expect(back, isNotNull);
      expect(back!.year, civil.year);
      expect(back.month, civil.month);
      expect(back.day, civil.day);
    });

    test('daf yomi bavli via Limudim', () {
      final limudim = Limudim();
      final daf = limudim.dafYomiBavli(2017, 12, 28);

      expect(daf, isNotNull);
      expect(daf!.tractate, TractateCode.shevuos);
      expect(daf.page, 30);
    });

    test('preset count via ZmanPresets', () {
      final presets = ZmanPresets();
      expect(presets.presetCount(), 167);
    });
  });
}

File: interop/packages/typescript/test/smoke.test.mjs
import assert from "node:assert/strict";
import test from "node:test";

import {
  Calendar,
  CivilDate,
  Limudim,
  TractateCode,
  ZmanPresets,
} from "../lib/index.mjs";

test("gregorian/hebrew round trip via Calendar", () => {
  const calendar = new Calendar();
  const civil = new CivilDate({ year: 2024, month: 1, day: 20 });

  const hebrew = calendar.gregorianToHebrew(civil);
  assert.ok(hebrew);
  assert.equal(hebrew.year, 5784);
  assert.equal(hebrew.month, 5);
  assert.equal(hebrew.day, 10);

  const back = calendar.hebrewToGregorian(hebrew);
  assert.ok(back);
  assert.equal(back.year, civil.year);
  assert.equal(back.month, civil.month);
  assert.equal(back.day, civil.day);
});

test("daf yomi bavli via Limudim", () => {
  const limudim = new Limudim();
  const daf = limudim.dafYomiBavli(2017, 12, 28);

  assert.ok(daf);
  assert.equal(daf.tractate, TractateCode.Shevuos);
  assert.equal(daf.page, 30);
});

test("preset count via ZmanPresets", () => {
  const presets = new ZmanPresets();
  assert.equal(presets.presetCount(), 167);
});
